citar.el

```diff
@@ -160,11 +170,59 @@ and nil means no action."
   :type '(choice (const :tag "Prompt" 'prompt)
                  (const :tag "Ignore" nil)))
 
+(defcustom citar-open-note-function
+  'citar-org-open-notes-default
+  "Function to open and existing or create a new note.
+
+A note function must take two arguments:
+
+KEY: a string to represent the citekey
+ENTRY: an alist with the structured data (title, author, etc.)
+
+If you use 'org-roam' and 'org-roam-bibtex', you can use
+'orb-bibtex-actions-edit-note' for this value."
+  :group 'citar
+  :type 'function)
+
+
 (defcustom citar-at-point-function 'citar-dwim
   "The function to run for 'citar-at-point'."
   :group 'citar
   :type 'function)
 
+(defcustom citar-major-mode-functions
+  '(((latex-mode) .
+     ((local-bib-files . citar-latex--local-bib-files)
+      (keys-at-point . citar-latex--keys-at-point)))
+    ((markdown-mode) .
+     ((local-bib-files . citar-markdown--local-bib-files)
+      (insert-keys . citar-markdown--insert-keys))))
+  "The variable determining the major mode specifc functionality.
```

Comment:
An entry for org-mode is needed here as well, isn’t it? Getting local bib-files previously depended on `bibtex-completion-find-local-bibliography` (which actually looked for file-extensions rather than major modes, which means it doesn’t work in org capture buffers), but now it should be enough with `org-cite-list-bibliography-files` here?
This seemed to work well (also in my org capture buffers):
```
((org-mode) .
         ((local-bib-files . org-cite-list-bibliography-files)))
```

Reply:
I hadn't actually tested that, but was assuming org handled it.

Are you aware of "suggested change" comments? There's ba like plus icon on the toolbar.

Reply:
PS - reminds me: should change the major mode check to use `derived-mode-p`.

Reply:
@andersjohansson - I added a little function to handle this `citar-org-local-bibs` and have hooked it up.

If you have time, could you please test?

I still have to write the documentation, hopefully sometime today, but am hoping to merge soon.

Reply:
> Are you aware of "suggested change" comments? There's a plus icon on the toolbar.
Will use that next time!

> that function returns global + local
Yes, that was my expectation, but then again, I may have missed the point of these settings.

My use-case is that I only use a global json-file (exported from zotero via betterbibtex) for all citing (and only cite in org) so I have:
```
(setq org-cite-global-bibliography '("BIBFILE.json")
      citar-bibliography nil)
```
Earlier, I experienced all entries were duplicated if I had both variables set to the same global file, but this doesn't seem to be the case anymore, so perhaps that (as recommended in the readme?) would be the right thing to do now then in any case (which means my problems are solved).

Reply:
What if you do this (or vice versa)?

```elisp
(setq citar-bibliography '("BIBFILE.json")
      org-cite-global-bibliography citar-bibliography)
```